### ISSUE TITLE
Support for Godot Type-Safe lines

### ISSUE THREAD

**Issue by emarino135** [godotengine/godot-vscode-plugin#436]
### Godot version

3.5.stable

### VS Code version

1.73.0

### Godot Tools VS Code extension version

1.3.1

### System information

Windows 10

### Problem statement

![image](https://user-images.githubusercontent.com/60202421/200151023-1165ee16-f9f9-46d9-92cb-3b93e254657c.png)
Similar to how Godot built in script editor supports Safe-Type lines by showing darkened numbers on the side of the editor

Safe-Type from Docs:
https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/static_typing.html

### Proposed solution

At the minimum I think some kind of terminal tree output with filenames and line numbers of unsafe lines would be great and get the job done.
Maybe a function we could call using command palette like 
> Godot-tool show unsafe lines

Dream Solution:
I don't know how the language Linter you used works to detech if line is unsafe but if it could detect it was unsafe and use a function like [text-marker extention](https://marketplace.visualstudio.com/items?itemName=ryu1kn.text-marker) to just highlight the line of code within vscode text editor that would work.

**Comment by DaelonSuzuka** [comment]
That's a somewhat interesting feature but I would be surprised if the language server provides the information required to implement it.

**Comment by chexmo** [comment]
this would be pretty useful.... it seems like (sadly) one can never get the best of the two worlds... text editor inside godot hasn't the power of vscode, but in vscode you lack of these kind of things, or the icons that hints there is a node signal connected, etc.

**Comment by geekley** [comment]
I would be happy if this could be added even as info blue squiggles on the line (that I could disable in a setting).

Unrelated, but... it would be also nice to have an option to render the "unused" warnings not as yellow warnings, but like how you get "unused" in other languages (less opacity).

**Comment by DaelonSuzuka** [comment]
> Unrelated, but... it would be also nice to have an option to render the "unused" warnings not as yellow warnings, but like how you get "unused" in other languages (less opacity).

I... think I know how to do that!

**Comment by geekley** [comment]
> > Unrelated, but... it would be also nice to have an option to render the "unused" warnings not as yellow warnings, but like how you get "unused" in other languages (less opacity).
> 
> I... think I know how to do that!

Yeah, should be easy, you probably just need to check the error codes containing "UNUSED" and add a [DiagnosticTag.Unnecessary](https://vscode-api.js.org/enums/vscode.DiagnosticTag.html#Unnecessary) to their diagnostic, right? Possibly also reduce the level to info or hint if that alone doesn't hide the yellow squiggles.

**Comment by Zylann** [comment]
That would be quite a nice insight to get this in VSCode. One example where it really helps is when you call a function that doesn't exist on a typed variable. That's actually a case where Godot will not report an error. It's valid GDScript and will fail at runtime, because of its dynamic nature.
The only way to catch this without finding it by chance when running the code, is by keeping an eye on type-safe lines, and spotting those that aren't safe but should. We can't see that in VSCode.
